api/util.py

```diff
@@ -115,8 +113,11 @@ def commit_file(dbc, _id, datainfo, filepath, data_path, force=False):
                 else: # existing file has different content
                     log.debug('Replacing   %s' % filename)
                     shutil.move(filepath, target_filepath)
+                    update_set = {'files.$.dirty': True, 'files.$.modified': datetime.datetime.utcnow()}
+                    for k,v in fileinfo.iteritems():
+                        update_set['files.$.' + k] = v
                     dbc.update_one({'_id':_id, 'files.filename': fileinfo['filename']},
-                            {'$set': {'files.$.dirty': True, 'files.$.modified': datetime.datetime.utcnow()}})
+                            {'$set': update_set})
```

Comment:
in this branch of the code, we are overriding an existing file.
```update_set``` allows to update all the file info like size, hash, etc.

This comes from this bugfix for the BIDS guys: https://github.com/scitran/api/commit/dc198a9b6010b8562d5f4c9170b5ab966ebab682